core/shared/src/main/scala/chartreuse/PlotModule.scala

```diff
@@ -94,13 +99,58 @@ trait PlotModule(numberFormat: NumberFormat) {
       val xTicksSequence = ticksToSequence(xTicks, scale, asX)
       val yTicksSequence = ticksToSequence(yTicks, scale, asY)
 
+      val xMajorTickToMinorTick: (ScreenCoordinate, Double, Int) => (
+          ScreenCoordinate,
+          DataCoordinate
+      ) = (screenCoordinate, interval, i) => {
+        val x = screenCoordinate.x - interval * i
+        (
+          ScreenCoordinate(x, 0),
+          DataCoordinate(x, 0)
+        )
+      }
+
+      val yMajorTickToMinorTick: (ScreenCoordinate, Double, Int) => (
+          ScreenCoordinate,
+          DataCoordinate
+      ) = (screenCoordinate, interval, i) => {
+        val y = screenCoordinate.y - interval * i
+        (
+          ScreenCoordinate(0, y),
+          DataCoordinate(0, y)
+        )
+      }
+
+      val toIntervalX: TicksSequence => Double = ticksSequence => {
+        val (screenCoordinate, _) = ticksSequence.head
+        val (screenCoordinate2, _) = ticksSequence.tail.head
```

Comment:
Please let me know if it's okay to name vals like this.

Reply:
1. You could have used more descriptive names
2. Use a pattern match to extract the elements
E.g. (untested)
```scala
val here +: neighbour +: _ = ticksSequence
```